Multitenancy.md

```diff
@@ -0,0 +1,226 @@
+
+When creating a wallet `wallet_dispatch_type` be used to specify how webhooks for the wallet should be dispatched. The options are:
+
+- `default`: Dispatch only to webhooks associated with this wallet.
+- `base`:` Dispatch only to webhooks associated with the base wallet.
+- `both`: Dispatch to both webhook targets.
+
+If either `default` or `both` is specified you can set the webhook urls specific to this wallet using the `wallet.webhook_urls` option.
+
+Example:
+
+```jsonc
+// POST /multitenancy/wallet
+{
+  // ... other params ...
+  "wallet_dispatch_type": "default",
+  "wallet_webhook_urls": [
+    "https://webhook-url.com/path",
+    "https://another-url.com/site"
+  ]
+}
+```
+
+### Identifying the wallet
+
+When the webhook urls of the base wallet are used or when multiple wallets specify the same webhook url it can be hard to identify the wallet an event belongs to. To resolve this each webhook event will include the wallet id the event corresponds to.
+
+For http events the wallet id is included as the `x-wallet-id` header. For WebSockets the wallet id is included in the enclosing JSON object.
+
+HTTP example:
+
+```jsonc
+POST <webhook-url>/{topic} [headers=x-wallet-id]
+{
+    // event payload
+}
+```
+
+WebSocket example:
+
+```jsonc
+{
+  "topic": "{topic}",
+  "wallet_id": "{wallet_id}",
+  "payload": {
+    // event payload
+  }
+}
+```
+
+## Authentication
+
+When multi-tenancy is not enabled you can authenticate with the agent using the `x-api-key` header. As there is only a single wallet, this provides sufficient ...
+
+For sub wallets an additional authentication method is introduced using Json Web Tokens (JWTs). A `token` parameter is returned after creating a wallet or calling the get token endpoint. This token must be provided for every admin API call you want to perform for the wallet using the Bearer authorization scheme.
+
+Example
+
+```
+GET /connections [headers="Authorization: Bearer {token}]
+```
+
+The `Authorization` header is in addition to the Admin API key. So if the `admin-api-key` is enabled (which it should be in production) both the `Authorization` and the `x-api-key` headers should be provided when making calls to a sub wallet. For calls to a base wallet only the `x-api-key` should be provided.
+
+### Getting a token
+
+A token can be obtained in two ways. The first method is the `token` parameter from the response of the create wallet (`POST /multitenancy/wallet`) endpoint. The second option is using the get wallet token endpoint (`POST /multitenancy/wallet/{wallet_id}/token`) endpoint.
+
+In umanaged mode the get token endpoint also requires the `wallet_key` parameter to be included in the request body. The wallet key will be included in the so the wallet can be unlocked when making request to the admin API.
```

Comment:
`The wallet key will be included in the` ? `so the wallet can be unlocked when marking request to the admin API`